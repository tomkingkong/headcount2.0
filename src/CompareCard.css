.CompareCard {
  color: white;
  display: flex;
  border-left: 0.3rem dotted white;
  border-right: 0.3rem dotted white;
  border-bottom: 0.3rem dotted white;
  justify-content: space-around;
  margin: 0 auto;
  height: 5rem;
  max-width: 17rem;
  text-align: center;
}

h4 {
  padding: 0 3rem;
  width: 2rem;
}

p {
 margin: 1rem auto;
}


.hide h4 {
  display: none;
}