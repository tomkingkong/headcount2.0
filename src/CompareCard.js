import React from 'react';
import PropTypes from 'prop-types';

import './CompareCard.css';

const CompareCard = ({comparedData}) => {
  const data = Object.keys(comparedData);
  return (
    <article className={typeof comparedData.compared === 'number' ? "CompareCard" : "CompareCard hide"}>
      <h4>{comparedData[data[0]]}<br/>AVG</h4>
      <p>{comparedData.compared}</p>
      <h4>{comparedData[data[1]]}<br/>AVG</h4>
    </article>
  );
};

CompareCard.propTypes = {
  comparedData: PropTypes.object
};

export default CompareCard;