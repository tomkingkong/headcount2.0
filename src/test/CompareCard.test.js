import React from 'react';
import { shallow } from 'enzyme';

import CompareCard from '../CompareCard';

const mockData =  {
  "comparedData": {
    'COLORADO': 0.53, 
    'ASPEN 1': 0.999,
    'compared': 0.531
  }
};

describe('CompareCard', () => {

  it('should match the snapshot passing an empty data object', () => {
    const wrapper = shallow(<CompareCard comparedData={{}}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot passing an array with data', () => {
    const wrapper = shallow(<CompareCard comparedData={mockData}/>);
    expect(wrapper).toMatchSnapshot();
  });

});