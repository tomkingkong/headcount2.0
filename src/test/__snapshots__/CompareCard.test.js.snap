// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompareCard should match the snapshot passing an array with data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CompareCard
    comparedData={
        Object {
            "comparedData": Object {
              "ASPEN 1": 0.999,
              "COLORADO": 0.53,
              "compared": 0.531,
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h4>
          Object {
                    "ASPEN 1": 0.999,
                    "COLORADO": 0.53,
                    "compared": 0.531,
                  }
          <br />
          AVG
</h4>,
        <p />,
        <h4>
          <br />
          AVG
</h4>,
      ],
      "className": "CompareCard hide",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            Object {
              "ASPEN 1": 0.999,
              "COLORADO": 0.53,
              "compared": 0.531,
            },
            <br />,
            "AVG",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "ASPEN 1": 0.999,
            "COLORADO": 0.53,
            "compared": 0.531,
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          "AVG",
        ],
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            <br />,
            "AVG",
          ],
        },
        "ref": null,
        "rendered": Array [
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          "AVG",
        ],
        "type": "h4",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4>
            Object {
                        "ASPEN 1": 0.999,
                        "COLORADO": 0.53,
                        "compared": 0.531,
                      }
            <br />
            AVG
</h4>,
          <p />,
          <h4>
            <br />
            AVG
</h4>,
        ],
        "className": "CompareCard hide",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              Object {
                "ASPEN 1": 0.999,
                "COLORADO": 0.53,
                "compared": 0.531,
              },
              <br />,
              "AVG",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "ASPEN 1": 0.999,
              "COLORADO": 0.53,
              "compared": 0.531,
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            "AVG",
          ],
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              <br />,
              "AVG",
            ],
          },
          "ref": null,
          "rendered": Array [
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            "AVG",
          ],
          "type": "h4",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`CompareCard should match the snapshot passing an empty data object 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CompareCard
    comparedData={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h4>
          <br />
          AVG
</h4>,
        <p />,
        <h4>
          <br />
          AVG
</h4>,
      ],
      "className": "CompareCard hide",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            <br />,
            "AVG",
          ],
        },
        "ref": null,
        "rendered": Array [
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          "AVG",
        ],
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            <br />,
            "AVG",
          ],
        },
        "ref": null,
        "rendered": Array [
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          "AVG",
        ],
        "type": "h4",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4>
            <br />
            AVG
</h4>,
          <p />,
          <h4>
            <br />
            AVG
</h4>,
        ],
        "className": "CompareCard hide",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              <br />,
              "AVG",
            ],
          },
          "ref": null,
          "rendered": Array [
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            "AVG",
          ],
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              <br />,
              "AVG",
            ],
          },
          "ref": null,
          "rendered": Array [
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            "AVG",
          ],
          "type": "h4",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
